packages/auto-approve/src/auto-approve.ts

```diff
@@ -56,6 +56,56 @@ export async function authenticateWithSecret(
   return new Octokit({auth: payload});
 }
 
+/**
+ * Grabs an etag for a PR, and retries adding a label if etag is different
+ *
+ * @param retryCount the amount of retries for adding a label
+ * @param owner string, of the repo of the incoming PR
+ * @param repo string, the name of the repo of the incoming PR
+ * @param prNumber number, the number of the PR
+ * @param octokit the octokit instance
+ */
+export async function retryAddLabel(
+  retryCount: number,
+  owner: string,
+  repo: string,
+  prNumber: number,
+  octokit: Octokit
```

Comment:
I would recommend injecting the GCFLogger instance into this function as an optional param that defaults to the global logger.
```suggestion
  octokit: Octokit,
  logger: GCFLogger = defaultLogger
```

Where `defaultLogger` is imported from `gcf-utils` like:
```ts
import {getContextLogger, logger as defaultLogger} from 'gcf-utils';
```

This will allow these log messages to keep the request context for easier filtering in the console.